lib/manager/cocoapods/readme.md

```diff
@@ -0,0 +1,232 @@
+In case of the repo you also can specify branch, tag or commit.
+
+```
+pod 'AFNetworking', '~> 2.6'
+pod 'Alamofire', :path => '~/Documents/Alamofire'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :branch => 'dev'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :tag => '3.1.1'
+pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :commit => '0f506b1c45'
+```
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+Cocoapods supports dependencies from git source and local path.
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+Package file use semver 2.0. ([details](https://semver.org))
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Versioning scheme supports logical operators:
+
+- `> 0.1` Any version higher than 0.1
+- `>= 0.1` Version 0.1 and any higher version
+- `< 0.1` Any version lower than 0.1
+- `<= 0.1` Version 0.1 and any lower version
+
+In addition to the logic operators CocoaPods has an optimistic operator `~>`:
+
+- `~> 0.1.2` Version 0.1.2 and the versions up to 0.2, not including 0.2 and higher
+- `~> 0.1` Version 0.1 and the versions up to 1.0, not including 1.0 and higher
+- `~> 0` Version 0 and higher, this is basically the same as not having it.
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Package manager used for applications only
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+I don't see such reason.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+There is a index service for cocoapods. Developers always search for pod on this [site](https://github.com/CocoaPods):
+
+Also, all podspec are stored in one [repo](https://github.com/CocoaPods/Specs)
+
+---
+
+#### Will users need the capability to specify a custom host/registry to look up? Can it be found within the package files, or within other files inside the repository, or would it require Renovate configuration?
+
+User can provide remove git url ([details](https://guides.cocoapods.org/using/the-podfile.html#from-a-podspec-in-the-root-of-a-library-repo))
+
+`pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'`
+
+---
+
+#### Do the package files contain any "constraints" on the parent language (e.g. supports only v3.x of Python) or platform (Linux, Windows, etc) that should be used in the lookup procedure?
+
+Cocoapods supports only mac platform.
+
+---
+
+#### Will users need the ability to configure language or other constraints using Renovate config?
+
+I would add an option to check dependencies with only specific swift version.
```

Comment:
Is that something that is not normally specified in the Podfile?

Reply:
Swift version is specified in the Xcode project, and cocoapods fetch pods based on that value.

Reply:
Is that Swift version committed to the repo? If so then it seems like something we will need to retrieve too.

Reply:
Swift version is committed to the repo, to Xcode project.

Reply:
OK, please note how we can detect and extract that. e.g. if we have a `Podfile`, can we locate the "other" file containing Swift version relative to it? In this case we don't plan to update the version, but we need to know it in order to look up the right pods.

Basically, we need to extract all the information necessary from the project to return the right pod results.

Reply:
Ok, it's clear that we need to know swift version. Must be some CLI tool for this purpose (I'll check it). But I don't think that we will be able to do it on linux machine.

Reply:
If the swift version is defined within a file in the repository, we should be able to extract it ourselves without use of other CLI tool.

Reply:
And that should be independent of operating system so perfectly doable with Linux

Reply:
We still need to work out how we can extract the Swift version from a repo.